src/ui/actions/timeline.ts

```diff
@@ -174,8 +174,18 @@ export function setZoomRegion(region: ZoomRegion): SetZoomRegionAction {
   return { type: "set_zoom", region };
 }
 
-export function seek(point: ExecutionPoint, time: number, hasFrames: boolean): UIThunkAction {
+export function seek(
+  point: ExecutionPoint,
+  time: number,
+  hasFrames: boolean,
+  pauseId?: PauseId
+): UIThunkAction {
   return () => {
-    ThreadFront.timeWarp(point, time, hasFrames);
+    const pause = pauseId !== undefined ? Pause.getById(pauseId) : undefined;
+    if (pause) {
+      ThreadFront.timeWarpToPause(pause);
```

Comment:
hmm, we seek to a lot of different types of places... Checking in advance if we have a pause first might be a bit cumbersome, but i can see why you need it now and we can revisit this over time.